storm-multilang/dotnet/src/main/dotnet/Dotnet.Storm.Adapter/GlobalStorm.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Dotnet.Storm.Adapter.Extensions;
+using Dotnet.Storm.Adapter.Messaging;
+using log4net;
+using Newtonsoft.Json;
+using System;
+using System.Text;
+
+namespace Dotnet.Storm.Adapter
+{
+    internal sealed class GlobalStorm
+    {
+        private readonly static ILog Logger = LogManager.GetLogger(typeof(GlobalStorm));
+
+        public static void Send(OutMessage message)
+        {
+            Console.WriteLine(message);
+            Console.WriteLine("end");
+        }
+
+        public static Message Receive<T>() where T : InMessage
+        {
+            try
+            {
+                string message = ReadMessage();
+
+                if (message.StartsWith("["))
+                {
+                    return JsonConvert.DeserializeObject<TaskIdsMessage>(message);
+                }
+                return JsonConvert.DeserializeObject<T>(message);
+            }
+            catch (Exception ex)
+            {
+                //we're expecting this shouldn't happen
+                Logger.Error($"Message parsing error: {ex.GetMessageTrace()}");
+            }
+
+            // just skip incorrect message
+            return null;
+        }
+
+        private static string ReadMessage()
+        {
+            StringBuilder message = new StringBuilder();
+            string line;
+            do
+            {
+                line = Console.ReadLine();
+
+                if (line == "end")
+                    break;
+
+                if (!string.IsNullOrEmpty(line))
+                {
+                    message.AppendLine(line);
+                }
+            }
+            while (true);
```

Comment:
This goes into an infinite loop and chews all the CPU when there are no messages - I debugged it and the ReadLine returns null if there is nothing on the command line - 
I would consider adding the following:
```
if (line == null)
{
  Thread.Sleep(10);
}
```
We give ourselves 10 milliseconds to retry if there was not a message in the queue - this will allow the CPU to process other instructions -

Reply:
Actually ReadLine sleeps until there is something in STDIN

Reply:
If the standard input device is the keyboard, the ReadLine method blocks until the user presses the Enter key. it is from here - https://msdn.microsoft.com/en-us/library/system.console.readline(v=vs.110).aspx

Reply:
But generally ReadLine blocks until EOL in pipeline